src/control/events/generic.go

```diff
@@ -58,3 +46,33 @@ func StrInfoToProto(si *StrInfo) (*sharedpb.RASEvent_StrInfo, error) {
 
 	return pbInfo, nil
 }
+
+// NewGenericEvent creates a generic RAS event from given inputs.
+func NewGenericEvent(id RASID, typ RASTypeID, sev RASSeverityID, msg string,
```

Comment:
Hmm. I don't think we need to fix it for this PR, but this is a bit awkward, and feels like a missed opportunity to DRY up the other event constructors. Perhaps as a followup you could do something more like the following, using `NewPoolSvcReplicasUpdateEvent` and your shutdown event as examples:

```go
// in ras.go:
func New(evt *RASEvent) *RASEvent {
	if evt == nil {
		evt = &RASEvent{
			Msg: "undefined RAS event",
		}
	}

	// Set some defaults as necessary
	if evt.Severity == 0 {
		evt.Severity = RASSeverityInfo
	}
	if evt.Timestamp == "" {
		evt.Timestamp = common.FormatTime(time.Now())
	}
	if evt.Hostname == "" {
		getHostName := func() string {
			hn, err := os.Hostname()
			if err != nil {
				return "failed to get hostname"
			}
			return hn
		}
		evt.Hostname = getHostName()
	}
	if evt.ProcID == 0 {
		evt.ProcID = uint64(os.Getpid())
	}
	if evt.forwarded == nil {
		evt.forwarded = atm.NewBool(false)
	}

	return evt
}

func NewPoolSvcReplicasUpdateEvent(hostname string, rank uint32, poolUUID string, svcReplicas []uint32, leaderTerm uint64) *RASEvent {
	return New(&RASEvent{
		Msg:       "DAOS pool service replica rank list updated",
		ID:        RASPoolRepsUpdate,
		Hostname:  hostname,
		Rank:      rank,
		PoolUUID:  poolUUID,
		Type:      RASTypeStateChange,
		Severity:  RASSeverityError,
		ExtendedInfo: &PoolSvcInfo{
			SvcReplicas:    svcReplicas,
			RaftLeaderTerm: leaderTerm,
		},
	})
}

	// Raise event on systemwide shutdown
	if pbReq.GetHosts() == "" && pbReq.GetRanks() == "" && pbReq.GetKill() {
		svc.events.Publish(events.New(&events.Event{
			ID:		events.RASSystemStop,
			Type:		events.RASTypeInfoOnly,
			Msg:		"System-wide shutdown requested",
			Rank:		uint32(system.NilRank),
		}))
	}
```